lib/companion_web/controllers/fallback_controller.ex

```diff
@@ -0,0 +1,20 @@
+defmodule CompanionWeb.FallbackController do
+  @moduledoc """
+  Translates controller action results into valid `Plug.Conn` responses.
+
+  See `Phoenix.Controller.action_fallback/1` for more details.
+  """
+  use CompanionWeb, :controller
+
+  def call(conn, {:error, %Ecto.Changeset{} = changeset}) do
+    conn
+    |> put_status(:unprocessable_entity)
+    |> render(CompanionWeb.ChangesetView, "error.json", changeset: changeset)
+  end
```

Comment:
Could there be any sensitive data in the changeset that we don't want to return in an error response?

Reply:
No I don't think so, everything in the changeset is supplied by the user of the API. The view also extracts the errors from the changeset and just returns that using https://hexdocs.pm/ecto/Ecto.Changeset.html#traverse_errors/2